File: app/src/main/java/com/maciega/bartosz/crawl/model/UrlsMapper.java
package com.maciega.bartosz.crawl.model;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;

import io.realm.RealmObject;

/**
 * Created by Bartosz on 2016-10-18.
 */

public class UrlsMapper {

    public static List<Url> convert(List<String> list) {
        List<Url> urlsList = new ArrayList<>();
        for (String urlString : list) {
            Url url = new Url();
            url.setUrl(urlString);
            urlsList.add(url);
        }

        return urlsList;
    }

    public static List<RealmObject> convert(Set<String> urls){
        List<RealmObject> urlList = new ArrayList<>();
        for(String url : urls){
            Url urlInstance = new Url();
            urlInstance.setUrl(url);
            urlList.add(urlInstance);
        }
        return urlList;
    }

}

File: app/src/main/java/com/maciega/bartosz/crawl/model/UrlsMapper.kt
package com.maciega.bartosz.crawl.model

import java.util.ArrayList

import io.realm.RealmObject

/**
 * Created by Bartosz on 2016-10-18.
 */

object UrlsMapper {

    fun convert(list: List<String>): List<Url> {
        val urlsList = ArrayList<Url>()
        for (urlString in list) {
            val url = Url()
            url.url = urlString
            urlsList.add(url)
        }

        return urlsList
    }

    fun convert(urls: Set<String>): List<RealmObject> {
        val urlList = ArrayList<RealmObject>()
        for (url in urls) {
            val urlInstance = Url()
            urlInstance.url = url
            urlList.add(urlInstance)
        }
        return urlList
    }

}
